AnalyticsTelemetricsData/AnalyticsTelemetricsData/Views/AnalyticsView.swift

```diff
@@ -0,0 +1,83 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import SwiftUI
+
+struct AnalyticsView: View {
+    var body: some View {
+        TrackEventView()
+            .onAppear {
+                Task {
+                    await AnalyticsManager.shared.configure()
+                }
+            }
+    }
+}
+
+struct TrackEventView: View {
+    @State var events: [String] = [String]()
+
+    @State var label: String = ""
+    @State var info: String = ""
+    @State var category: AnalyticsType = .event
+
+    var body: some View {
+        VStack {
+            VStack {
+                Form {
+                    Section {
+                        TextField("Event Label", text: $label)
+                        TextField("Extra Info", text: $info)
+                        Picker("Event Type", selection: $category) {
+                            ForEach(AnalyticsType.allCases) {
+                                Text("\($0.rawValue)")
+                            }
+                        }
+                    }
+                    Section {
+                        HStack {
+                            Spacer()
+                            Button("Track") {
+                                // Log an analytics event
+                                AnalyticsManager.shared.logEvent(category, label: label, info: ["extra": info])
+                                events.append(label)
+                                label = ""
+                                info = ""
+                                category = .event
+                            }
+                            .buttonStyle(.bordered)
+                            .cornerRadius(4)
+                            // Log a click event in SwiftUI
+                            .logClick("track_click")
+                            Spacer()
+                        }
+                    }
+                }
+                Spacer()
+                List(events, id: \.self) { event in
+                    Text(event)
```

Comment:
The events we are showing in this list are the events created using the form, all the other events like screen views, button clicks are not showed on this list. Because asymmetric objects cannot be queried, we don't have access to the events sent on the current session/ever unless we stored them somehow. If this list is too confusing I can remove it @dacharyc

Reply:
The form has a `Picker` that includes all of the `AnalyticsType` cases, so I assumed I was meant to be creating events of different types I could see on the backend. But the list only showed the label, not the event type. So I could add a `.click` or `.screen_view` event with the current form implementation, and the label would show up in the list, but not the event type.

If we do _not_ intend folks to use the other `AnalyticsType` cases - only `.event` - then maybe we should just remove the `Picker` from the form and this isn't really an issue.

If we do intend folks to add different event types just as a way to populate test data, we could add an intermediary struct to populate the list - I went with this in my testing:

```swift
struct EventListEvent: Hashable {
    var id = UUID()
    var label = ""
    var eventType: AnalyticsType
}
```

And then switch the `@State var events` from an array of string labels to an array of this struct. I did this:

```swift
@State var events: [EventListEvent] = []
```

Then I updated the `Button` action to:

```swift
events.append(EventListEvent(label: label, eventType: category))
```

And then made the List an HStack to also display the category:

```swift
 HStack {
     Text(event.label)
     Spacer()
     Text(event.eventType.rawValue)
}
```

Reply:
Thanks for the suggestion. I didn't check all the code and I though we were only tracking events of event type "event"